facebook/react-native
# update @react-native-community/eslint-config

## Summary

I updated **dependencies** for `@react-native-community/eslint-config` and adjusted configs to align with updated versions.
### Main changes:

1. `babel-eslint` became [`@babel/eslint-parser`](https://github.com/babel/babel/tree/main/eslint/babel-eslint-parser)
2. a companion (recommended) [`@babel/eslint-plugin`](https://github.com/babel/babel/tree/main/eslint/babel-eslint-plugin) is used for `@babel/semi` rule (handles class properties)
3. `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` were updated to support new TypeScript syntax
4. `prettier` were also updated
5. `prettier/react` was removed according to [eslint-config-prettier](https://github.com/prettier/eslint-config-prettier) documentation


## Changelog
## Test Plan

1. Create new react-native project
2. Add updated `@react-native-community/eslint-config` into it (`yarn add file://....`)
3. Run `yarn lint`
4. Check there is no errors or warnings
